paasta_tools/marathon_tools.py

```diff
@@ -214,30 +214,35 @@ def get_instances(self):
                   specified or if desired_state is not 'start'.
                   """
         if self.get_desired_state() == 'start':
-            if self.get_max_instances() is not None:
-                try:
-                    zk_instances = get_instances_from_zookeeper(
-                        service=self.service,
-                        instance=self.instance,
-                    )
-                    log.debug("Got %d instances out of zookeeper" % zk_instances)
-                except NoNodeError:
-                    log.debug("No zookeeper data, returning max_instances (%d)" % self.get_max_instances())
-                    return self.get_max_instances()
-                else:
-                    limited_instances = self.limit_instance_count(zk_instances)
-                    if limited_instances != zk_instances:
-                        log.warning("Returning limited instance count %d. (zk had %d)" % (
-                                    limited_instances, zk_instances))
-                    return limited_instances
-            else:
-                instances = self.config_dict.get('instances', 1)
-                log.debug("Autoscaling not enabled, returning %d instances" % instances)
-                return instances
+            return self.get_instances()
         else:
             log.debug("Instance is set to stop. Returning '0' instances")
             return 0
```

Comment:
This function and `PaastaNativeServiceConfig.get_instances()` are now very similar -- can you put this function in `LongRunningServiceConfig`, make `PaastaNativeServiceConfig.get_instances()` just `return self.config_dict.get('instances')`, and change the two `get_instances` calls in native_mesos_scheduler to `get_desired_instances()`?